python/lsst/dax/imgserv/locateImage.py

```diff
@@ -53,51 +53,105 @@
 class W13Db:
     '''Base class for examining DC_W13_Stripe82 data
     '''
-    def __init__(self, credFileName, database, table, columns, dataRoot, logger):
+    def __init__(self, credFileName, database, table, columns, dataRoot, butlerPolicy, butlerKeys, logger):
         self._log = logger
         self._table = table
         self._columns = columns
         self._conn = getEngineFromFile(credFileName, database=database).connect()
         self._dataRoot = dataRoot
+        self._butlerPolicy = butlerPolicy
+        self._butlerKeys = butlerKeys
         sql = "SET time_zone = '+0:00'"
         try:
             self._log.info(sql)
             self._conn.execute(sql)
         except SQLAlchemyError as e:
             self._log.error("Db engine error %s" % e)
 
-    def getImageFull(self, ra, dec):
-        '''Return an image containing ra and dec.
+    def getMdPolicy(self):
+        '''Return the butler policy name to retrieve metadata
+        '''
+        return self._butlerPolicy + "_md"
+
+    def getIdsFromRequest(self, request):
+        '''Returns a dictionary of key value pairs from the request with
+        valid being true if there were entries for everything in _butlerKeys.
+        This will not work for floating point values.
+        '''
+        valid = True
+        ids = {}
+        for key in self._butlerKeys:
+            value = request.args.get(key)
+            if value == None:
+                valid = False
+            try:
+                value = int(value)
```

Comment:
In theory, it's possible for a dataId to use strings containing digits, floats, or something else.  But I think this is a reasonable limitation for now.